diff --git a/components/Image.astro b/components/Image.astro
index b84bcacf7a41cb7842016c789e3a75132b1348f0..9539ef7e8051f9fe18cb7d600bc51589a4732e61 100644
--- a/components/Image.astro
+++ b/components/Image.astro
@@ -56,12 +56,17 @@ if (!name) {
 if (!props.width && !props.height) {
     props.width = props.height = size;
 }
-
+const isSvg = url.endsWith(".svg");
+let svgHtml = "";
 // Download only on production to avoid downloading multiple times
 if (process.env.NODE_ENV === "production" && !url.startsWith("/")) {
     try {
         const response = await fetch(url);
         const blob = await response.blob();
+        svgHtml = await response.text();
+        console.log(
+            "got some response as svg: ", svgHtml
+        )
         fs.mkdirSync("public/assets/preloaded", { recursive: true });
         fs.writeFileSync(
             `public/assets/preloaded/${name}`,
@@ -79,4 +84,13 @@ if (process.env.NODE_ENV === "production" && !url.startsWith("/")) {
 }
 ---
 
-<img {...props} src={url} alt={alt} />
+{isSvg? 
+(
+    <div set:html={svgHtml}/>
+)
+:
+(
+    <img {...props} src={url} alt={alt} />
+)
+
+}
